src/irr/video/COpenGLRenderpassIndependentPipeline.h

```diff
@@ -14,6 +14,63 @@ class COpenGLRenderpassIndependentPipeline : public IGPURenderpassIndependentPip
 public:
     using IGPURenderpassIndependentPipeline::IGPURenderpassIndependentPipeline;
 
+    struct SVAOHash
+    {
+        constexpr static size_t getHashLength()
+        {
+            return sizeof(hashVal)/sizeof(uint32_t);
+        }
+
+        bool operator!=(const SVAOHash& rhs) const 
+        {
+            for (size_t i = 0u; i < getHashLength(); ++i)
+                if (hashVal[i] != rhs.hashVal[i])
+                    return true;
+            return false;
+        }
+        bool operator==(const SVAOHash& rhs) const
+        {
+            return !((*this) != rhs);
+        }
+
+        uint32_t getRelativeOffsetForAttrib(uint32_t _attr) const
+        {
+            if (_attr == 5u)
+            {
+                uint32_t val = (relOffsets[0] & (0xfu<<60)) >> 60;
+                val |= (relOffsets[1] & 0xffu) << 4;
+                return val;
+            }
+            if (_attr == 11u)
+            {
+                uint32_t val = (relOffsets[1] & (0xffu<<56)) >> 56;
+                val |= (relOffsets[2] & 0xfu) << 8;
+                return val;
+            }
+            const uint32_t ix = (_attr > 5u) + (_attr > 11u);
+            const uint32_t shift = (_attr*12u) - (ix*64u);
+
+            return (relOffsets[ix]>>shift) & 0xfffu;
+        }
+
+        union {
+#include "irr/irrpack.h"
+            struct {
+                //can't find any info about what's guaranteed "min max" for rel offset, but here i'm assuming 2048 (same as with stride)
+                uint64_t relOffsets[3];
+                //GL's guaranteed minimal for max stride is 2048 so 12 bits per attrib needed
+                //TODO do we want to include stride in hash? Stride is parameter of binding, while relative offset is parameter of attribute
+                uint64_t strides[3];//16*12 bits
+                uint64_t mapAttrToBinding;//16*4 bits
+                uint16_t enabledAttribs;
```

Comment:
you can put this in `attribFormatAndComponentCount` as special value `ECF_UNKNOWN`

Reply:
@Crisspl done?

Reply:
yea